### DISCUSSION TITLE
Spring boot3.3, graphql-java22 -> "Variable 'xyz' has invalid value: Expected a Boolean input but, it was a 'String'

### DISCUSSION THREAD

**Discussion by srikanthinu** [graphql-java/graphql-java#3690] [Q&A]
We are upgrading grapqhl from 19 to 22 but seeing this error for all primitive (boolean and int). Further digging into it I see error is from GraphqlBooleanCoercing.parseValueImpl -> which just check instance of boolean and throw exception

BTW env -> Spring boot 3.3, graphql 22

**Comment by bclozel** [comment]
Unfortunately it’s hard to help you with so little information.

Can you share a complete stacktrace? A minimal sample application as a zip file or GitHub repo would also help.

**Comment by dondonz** [comment]
Hello (from the future!)

You probably already solved your problem, but for anyone else who sees an error message like this, it may be due to changes in `parseValue` coercion in v22.0. New documentation is on the way: https://github.com/graphql-java/graphql-java-page/pull/186/files
